mesh/mesh.go

```diff
@@ -276,18 +342,34 @@ func (m *Mesh) GetLatestView() []types.BlockID {
 
 func (m *Mesh) AddBlock(blk *types.Block) error {
 	m.Debug("add block %d", blk.ID())
+	m.AtxDB.ProcessBlockATXs(blk) // change this to return error if process failed
 	if err := m.MeshDB.AddBlock(blk); err != nil {
 		m.Error("failed to add block %v  %v", blk.ID(), err)
 		return err
 	}
-	m.AtxDB.ProcessBlockATXs(blk)
 	m.SetLatestLayer(blk.Layer())
 	//new block add to orphans
 	m.handleOrphanBlocks(blk)
 	//m.tortoise.HandleLateblock.Block(block) //why this? todo should be thread safe?
 	return nil
 }
 
+//todo this overwrites the previous value if it exists
+func (m *Mesh) AddLayer(layer *types.Layer) error {
+	if len(layer.Blocks()) == 0 {
+		m.layers.Put(layer.Index().ToBytes(), []byte{})
+		return nil
+	}
+
+	//add blocks to mDB
+	for _, bl := range layer.Blocks() {
```

Comment:
this can cause only some of the blocks in the layer to be written, and some will not if one block in `layer` could not be added. add a note about it since it is a dangerous situation we need to think about

Reply:
used only in tests and returns error